src/DiffSharp.Core/Tensor.fs

```diff
@@ -567,7 +577,7 @@ type Tensor =
     member t.SumT2Dim0() = Tensor.SumT2Dim0(t)
     
     static member Transpose (a:Tensor) =
-        if a.Dim <> 2 then failwithf "Expecting a 2d Tensor, received Tensor with shape %A" a.Shape
+        if a.Dim > 2 then failwithf "Expecting at least a 2d tensor, received Tensor with shape %A" a.Shape

```

Comment:
Are we sure this is correct?

Reply:
No it's not correct, thanks!  Will make sure we test Transpose directly on batching